### ISSUE TITLE
Meaning of the `catching_up` field in RPC

### ISSUE THREAD

**Issue by areller** [tendermint/tendermint#7141]
I run a tendermint node in Kubernetes and I need to create a readiness check that accurately tells me that 

* the node is functioning and there are no errors
* the node is synced up with the latest blockchain state

up until now, I was relying on calling `POST /status`, verifying that I get a status 200, and checking that `catching_up` is `false`

But then there was an incident when a node got stuck on a certain block height, because it required a version upgrade, but `catching_up` was reporting as `false`, causing the readiness check to report that the node is functioning.  

My guess is that `catching_up` only indicates that the node is currently trying to sync with other peers, but if there's an issue that prevents it from doing so, it will report `false` 
I couldn't find any documentation about that so I'm just guessing for now.

So my questions are,

* What's the exact meaning of the `catching_up` field, is my idea of it accurate?
* What's the best way to check from RPC or command line (`cosmovisor`) that the node is indeed working and up to date (synced up fully with latest blockchain state)

**Comment by cmwaters** [comment]
Hi, `catching_up` is set to `true` only when either `blocksync` or `statesync` are in use and `consensus` is therefore waiting for either of these processes to complete (i.e. reach a certain state close to the head of the blockchain). 

The consensus mechanism is also capable of catching up nodes to the head of the blockchain (although much slower) and thus it's possible that neither `blocksync` or `statesync` are running and the node is still behind the head of the blockchain. Additionally, when there is an upgrade and the chain halts at a specific height, consensus will still be running, although unable to make any progress and thus `catching_up` will still be false. In the future, we will introduce better semantics to indicate an upgrade.

In summary, `catching_up` should be considered a rough metric of the state of the node and there are probably a few things we could do to make it more accurate (although low-priority). I think the best way to find out whether your chain is up to date is to query a few nodes and see how far off your height is from them. This is actually done in `blocksync` and is displayed as `max_peer_block_height` in `/status`. Hope that clarified things.
